src/include/mpir_comm.h

```diff
@@ -195,6 +195,10 @@ struct MPIR_Comm {
 
     int revoked;                /* Flag to track whether the communicator
                                  * has been revoked */
+    int seq;                    /* A sequence number used for e.g. vci hashing.
+                                 * If seq==0, then every processes' seq is 0 */
+    int *seq_table;             /* Table of sequence numbers for each rank.
+                                 * If seq==0, seq_table is NULL */
```

Comment:
Could you remind me the motivation to add the `seq` field? I was trying to understand how different it will be from `comm->context_id` and `comm_recvcontext_id`.

Reply:
`context_id` is unique per communicator, including subcomms. Sub-communicators will have different context_id from their parents. Also context_id can be non-contiguous. If we directly use context_id to hash to vci, the vci may also get gaps, results in inefficiency.

`comm_recvcontext_id` is used to identify remote comm in a inter-communicator -- but I am not really familiar with it.

Reply:
Before merging, could you add this motivation to the code comment?

Reply:
OK

Reply:
done.